sqlitecluster/SQLiteNode.cpp

```diff
@@ -2609,13 +2605,22 @@ void SQLiteNode::handleRollbackTransaction(SQLite& db, Peer* peer, const SData&
 }
 
 SQLiteNode::State SQLiteNode::leaderState() const {
-    lock_guard<mutex> leadPeerLock(_leadPeerMutex);
+    shared_lock<shared_mutex> leadPeerLock(_leadPeerMutex);
     if (_leadPeer) {
         return _leadPeer.load()->state;
     }
     return State::UNKNOWN;
 }
 
+string SQLiteNode::leaderCommandAddress() const {
+    shared_lock<shared_mutex> leadPeerLock(_leadPeerMutex);
+    if (_leadPeer && _leadPeer.load()->state == State::LEADING) {
+        if (_leadPeer.load()->state == State::LEADING) {
+            return _leadPeer.load()->commandAddress;
+        }
+    }
```

Comment:
The nested `if` statement is redundant, right?

```suggestion
    if (_leadPeer && _leadPeer.load()->state == State::LEADING) {
        return _leadPeer.load()->commandAddress;
    }
```